
From: Wim Van Sebroeck <wim@iguana.be>

<wim@iguana.be> (04/06/27 1.1770)
   [WATCHDOG] indydog.c-patch-20040627
   
   * Fix: since we use the new module_param's: make sure that
   linux/moduleparam.h stays included
   * in the release code we can just use indydog_stop();

The ChangeSets can also be looked at on:
	http://linux-watchdog.bkbits.net:8080/linux-2.6-watchdog


Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/char/watchdog/indydog.c |   10 ++++------
 1 files changed, 4 insertions(+), 6 deletions(-)

diff -puN drivers/char/watchdog/indydog.c~v267-indydogc-patch-20040627 drivers/char/watchdog/indydog.c
--- 25/drivers/char/watchdog/indydog.c~v267-indydogc-patch-20040627	2004-06-27 22:39:41.204387680 -0700
+++ 25-akpm/drivers/char/watchdog/indydog.c	2004-06-27 22:39:41.208387072 -0700
@@ -12,6 +12,7 @@
  */
 
 #include <linux/module.h>
+#include <linux/moduleparam.h>
 #include <linux/config.h>
 #include <linux/types.h>
 #include <linux/kernel.h>
@@ -87,12 +88,9 @@ static int indydog_release(struct inode 
 {
 	/* Shut off the timer.
 	 * Lock it in if it's a module and we defined ...NOWAYOUT */
-	if (!nowayout) {
-		u32 mc_ctrl0 = sgimc->cpuctrl0;
-		mc_ctrl0 &= ~SGIMC_CCTRL0_WDOG;
-		sgimc->cpuctrl0 = mc_ctrl0;
-		printk(KERN_INFO "Stopped watchdog timer.\n");
-	}
+	if (!nowayout)
+		indydog_stop();		/* Turn the WDT off */
+
 	indydog_alive = 0;
 
 	return 0;
_
